beater/server.go

```diff
@@ -38,12 +38,9 @@ func newMuxer(config Config, report reporter) *http.ServeMux {
 	mux := http.NewServeMux()
 
 	for path, p := range processor.Registry.Processors() {
-
-		handler := createHandler(p, config, report)
-
+		handler := appHandler(p, config, report)
 		logp.Info("Path %s added to request handler", path)
-
-		mux.HandleFunc(path, handler)
+		mux.Handle(path, authHandler(config.SecretToken, handler))
 	}
 
 	mux.HandleFunc("/healthcheck", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This change just made me realise that we don't check the token for the `/healthcheck` handler I think?

Reply:
This looks fine for me.